src/views/pages/dashboard/invoice/DashboardInvoicePageTs.ts

```diff
@@ -150,14 +128,45 @@ export class DashboardInvoicePageTs extends Vue {
       return null
     }
   }
+
+  /**
+   * The transaction object
+   * @type {TransferTransaction | undefined}
+   */
+  get currentTransaction(): TransferTransaction | undefined {
+    if (this.shiftTransaction) {
+      return this.shiftTransaction.transaction as TransferTransaction
+    }
+  }
+
+  /**
+   * The first transaction in the list stack
+   * @type {ITransaction}
+   */
+  get shiftTransaction(): ITransaction {
+    if (this.transactions.length) {
+      return this.transactions.shift()
```

Comment:
if you call this method 2 times, it will return different results but it should not. Please use `this.transactions[0]` instead.